app/_src/kubernetes-ingress-controller/guides/using-ingress-with-grpc.md

```diff
@@ -86,3 +88,174 @@ talks to the upstream service:
     ```bash
     grpcurl -v -d '{"greeting": "Kong Hello world!"}' -insecure $PROXY_IP:443 hello.HelloService.SayHello
     ```
+{% endif_version %}
+
+{% if_version gte:2.9.x %}
+## Overview
+
+This guide walks through deploying a simple [Service][svc] that listens for
+[gRPC connections][gRPC] and exposes this service outside of the cluster using
+{{site.base_gateway}}.
+
+For this example, you will:
+* Deploy a gRPC test application.
+* Route gRPC traffic to it using Ingress or GRPCRoute.
+
+{% include_cached /md/kic/installation.md kong_version=page.kong_version %}
+
+{% include_cached /md/kic/class.md kong_version=page.kong_version %}
+
+[svc]:https://kubernetes.io/docs/concepts/services-networking/service/
+[gRPC]:https://grpc.io/
+
+## Prerequisite
+
+To make `gRPC` requests, you need a client that can invoke gRPC requests.
+In this guide, we use
+[`grpcurl`](https://github.com/fullstorydev/grpcurl#installation).
+Ensure that you have it installed on your local system.
+
+## Deploy a gRPC test application
+
+Add a gRPC deployment and service:
+
+```bash
+echo "---
+apiVersion: v1
+kind: Service
+metadata:
+  name: grpcbin
+  labels:
+    app: grpcbin
+spec:
+  ports:
+  - name: grpc
+    port: 443
+    targetPort: 9001
+  selector:
+    app: grpcbin
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: grpcbin
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: grpcbin
+  template:
+    metadata:
+      labels:
+        app: grpcbin
+    spec:
+      containers:
+      - image: moul/grpcbin
+        name: grpcbin
+        ports:
+        - containerPort: 9001
+" | kubectl apply -f -        
+```
+Response:
+```text
+deployment.apps/grpcbin created
+service/grpcbin created
+```
+
+## Route GRPC traffic
+
+Now that the test application is running, you can create GRPC routing configuration that
+proxies traffic to the application:
+
+{% navtabs api %}
+{% navtab Ingress %}
+```bash
+echo "apiVersion: networking.k8s.io/v1
+kind: Ingress
+metadata:
+  name: demo
+  annotations:
+  kubernetes.io/ingress.class: kong
+spec:
+  rules:
+  - http:
+    paths:
+    - path: /
+      backend:
+        serviceName: grpcbin
+        servicePort: 9001" | kubectl apply -f -
```

Comment:
Isn't the service above defining port 443 🤔 ?

Reply:
nice catch! copy-pasta error :)